### ISSUE TITLE
LDAP groups not working correctly

### ISSUE THREAD

**Issue by d-tamm** [owncloud/core#12306]
### Steps to reproduce
Connect OC to an openLDAP server as user backend (slapd  (Ubuntu) (Mar 17 2014 21:20:08), part of Samba installation).
Group filter: only "posixGroup"
Advanced -> Directory settings: Group member association "memberUid"
Expert -> UUID Attr for users: "uid"

### Expected behaviour
All LDAP users should have their corresponding LDAP groups. The users interface should show all LDAP users with their corresponding group membership according to LDAP. This was working correctly in OC 7.0.2.

### Actual behaviour
Entering the users module, all users are shown, but each user has only one group in the groups column, regardless of how many groups he/she really is a member of in LDAP. However, the groups view in the left part of the screen shows all LDAP groups with the right number of members. Clicking on one of the group names will correctly show the members. So this part seems to work.

All our users are in one main group which is used for sharing calendars. After upgrading our production environment from 7.0.2 to 7.0.3, most users got that main group displayed as their sole group in the users management module. However, one user got "Domain Admins" from LDAP as his only group. This user also lost access to all calendars shared with the main group.
However, in a test environment setup from scratch, this behaviour could not be reproduced.

I can also observe that the oc_ldap_group_members table in the database is empty in my test environment. The user_mapping and group_mapping tables are populated.

### Server configuration
**Operating system**: Ubuntu server 14.04

**Web server:** Apache

**Database:** mySQL

**PHP version:** 5.5.9

**ownCloud version:** 7.0.3

**Updated from an older ownCloud or fresh install:** Update via apt-get from 7.0.2, see even above

**List of activated apps:** Calendar, Contacts, Deleted files, Documents, First Run Wizard, Full Text Search, LDAP user and group backend, Pictures, Share Files, Updater, Versions

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'occ537937d2f',
  'passwordsalt' => 'xxx',
  'trusted_domains' => 
  array (
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '7.0.2.1',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => 'xxx',
  'installed' => true,
  'ldapIgnoreNamingRules' => false,
  'forcessl' => true,
  'theme' => '',
  'maintenance' => false,
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
No

### Client configuration
**Browser:** Firefox 33.0

**Operating system:** Ubuntu 14.04

#### Web server error log
Only some rows like this one:
[Wed Nov 19 17:10:51.028061 2014] [access_compat:error] [pid 12111] [client 127.0.0.1:37107] AH01797: client denied by server configuration: /var/www/owncloud/data/htaccesstest.txt

#### ownCloud log (data/owncloud.log)
{"app":"PHP","message":"ldap_search(): Partial search results returned: Sizelimit exceeded at \/var\/www\/owncloud\/apps\/user_ldap\/lib\/ldap.php#252","level":3,"time":"2014-11-19T16:08:39+00:00"}

**Comment by MorrisJobke** [comment]
@blizzz Did you have any clue what could cause this drop of group assignements if a user is in more than ne group?

**Comment by oxivanisher** [comment]
I am experiencing the same issue also. If i can help with debug information or anything, feel free to ask.

**Comment by tkaufm** [comment]
@blizzz 

as additional comment to https://github.com/owncloud/core/issues/13174: 
i run the script mention in here (https://gist.github.com/blizzz/736339dc4938a920e087). 
it seems like its only working for oc702 and highter but anyway.

On all oc6-versions i get a 'Invalid UserID'.
**oc700:** PHP Fatal error:  Call to undefined method OC\Server::getGroupManager() in /var/www/oc700/getUserGroupMemberships.php on line 21
**oc701:** PHP Fatal error:  Call to undefined method OC\Server::getGroupManager() in /var/www/oc701/getUserGroupMemberships.php on line 21
**oc702:** works and all groups are listed

```
# sudo -u www-data php -f ./getUserGroupMemberships.php arp rt
arp is member of following groups:
football
login
rt
all
social
tt

rt has following members:
ami (ami)
arp (arp)
awi (awi)
```


**oc703 & 704:** user is only listed with one groupmembership but the membersips from group to users are correct.
```
# sudo -u www-data php -f ./getUserGroupMemberships.php arp rt
arp is member of following groups:
social

rt has following members:
ami (ami)
arp (arp)
awi (awi)
...
```

Thanks

**Comment by blizzz** [comment]
@MorrisJobke yes, i think so, stay tuned.

**Comment by blizzz** [comment]
Could you all who are affected please try https://github.com/owncloud/core/pull/13280 ? Thank you!

**Comment by MorrisJobke** [comment]
Fix will be in 7.0.5 and 8.0.0
